client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala

```diff
@@ -719,6 +796,77 @@ class DeltaSharingRestClient(
     )
   }
 
+  private def getQueryInfoTargetUrl(table: Table, queryId: String) = {
+    val shareName = URLEncoder.encode(table.share, "UTF-8")
+    val schemaName = URLEncoder.encode(table.schema, "UTF-8")
+    val tableName = URLEncoder.encode(table.name, "UTF-8")
+    val encodedQueryId = URLEncoder.encode(queryId, "UTF-8")
+    val target = getTargetUrl(
+      s"/shares/$shareName/schemas/$schemaName/tables/$tableName/queries/$encodedQueryId")
+    target
+  }
+
+  private def getQueryInfo(
```

Comment:
Is this getQueryInfo or queryView? or asyncQueryTable?

Reply:
it's getQueryInfo, this where we pool the query status using the new getQueryInfo api.